gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -462,7 +462,11 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
                 plan.resultExpressions,
                 plan.child
               )
-            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+            val allTransformable = ColumnarPullOutProject
+              .getTransformedPlan(transformer)
+              .map(_.asInstanceOf[GlutenPlan].doValidate())
+              .reduce(ValidationResult.merge)
```

Comment:
I think we should only validate the aggregate, so we can decouple the fallback for project and aggregate.

Reply:
However, currently we can only obtain the Pre`ProjectExecTransformer` and Post`ProjectExecTransformer` generated by the Columnar Rule at this point. We cannot handle them separately like we do for verifying regular `ProjectExecTransformer`. I think we need to add a hint tag in the agg to record its pre/post-project fallback requirements, and then directly generate a fallback plan when generating pre/post-project.

Reply:
It should be easy to get the aggregate after applying pull out project rules. We pass an aggregate in and it should only output two options, 1) aggregate, 2) project-aggregate. So we can strip the top project if exists.